core/src/com/unciv/models/translations/Translations.kt

```diff
@@ -231,6 +232,14 @@ class Translations : LinkedHashMap<String, TranslationEntry>() {
         const val conditionalUniqueOrderString = "ConditionalsPlacement"
         const val shouldCapitalizeString = "StartWithCapitalLetter"
         const val effectBeforeCause = "EffectBeforeCause"
+
+        // NumberFormat cache, key: language, value: NumberFormat
+        private val languageToNumberFormat = mutableMapOf<String, NumberFormat>()
+
+        fun getNumberFormatFromLanguage(language: String): NumberFormat =
+            languageToNumberFormat.getOrPut(language) {
+                NumberFormat.getInstance(UncivGame.Current.settings.getLocaleFromLanguage(language))
+            }
```

Comment:
The function for "getLocaleFromLanguage" should be in Translations, currently it's all spaghetti - in gamesettings we call getnumberformat from translations which calles getlocale from settings

And why do we need the number formats cached in 2 separate places?

Reply:
> The function for "getLocaleFromLanguage" should be in Translations

Not sure what you meant. Isn't it already in companion object of Translations class.

Reply:
> And why do we need the number formats cached in 2 separate places?

Well... That's a good point... I actually forgot why I did that... fixed in 495bdd1